lib/pac_conn

```diff
@@ -904,11 +904,39 @@ sub _setSshProxy {
         if ($$CFG{'environments'}{$UUID}{'jump key'}) {
             print CFG "IdentityFile $$CFG{'environments'}{$UUID}{'jump key'}\n";
         }
-        print CFG qq“\nHost server\nHostName $IP\nUser $PASSPHRASE_USER\nPort $PORT\n“;
-        if (($PUBKEY)&&($$CFG{'environments'}{$UUID}{'auth type'} eq 'publickey')) {
-            print CFG "IdentityFile $PUBKEY\n";
+        if ($method =~ /ssh/i) {
+            # Second server is only important for SSH JumpHost
+            print CFG qq“\nHost server\nHostName $IP\nUser $PASSPHRASE_USER\nPort $PORT\n“;
+            if (($PUBKEY)&&($$CFG{'environments'}{$UUID}{'auth type'} eq 'publickey')) {
+                print CFG "IdentityFile $PUBKEY\n";
+            }
         }
         close CFG;
+        if ($method =~ /VNC|RDP/i) {
+            my $LPORT = $PORT;
+            my $break = 10;
+
+            $proxy_jump = 2;
+
+            if (-e "$CFG_DIR/tmp/$UUID.ctl") {
+                # Avoid open a second terminal with the same UUID
+                my $check = `ssh -F $CFG_DIR/tmp/$UUID.conf -S $CFG_DIR/tmp/$UUID.ctl -T -O check jumpserver 2>&1`;
+                if ($check =~ /=\d+/) {
+                    ctrl("UNHIDE_TERMINAL");
+                    die "$COLOR{'err'}Can not duplicate VNC/RDP connections$COLOR{'norm'}\nConfigure a different second connection.\n\n";
+                }
+            }
+            # Check if port we want to bind is not alreay in use
+            while (($break)&&(my $msg = `netstat -lnt | grep $LPORT 2>&1`)) {
+                if ($msg !~ /:$LPORT/) {
+                    # netstat not installed?
+                    last;
+                }
+                $LPORT++;
+                $break--;
+            }
+            system "ssh -F $CFG_DIR/tmp/$UUID.conf -S $CFG_DIR/tmp/$UUID.ctl -f -N -T -M -L $LPORT:localhost:$PORT jumpserver";
```

Comment:
Should be $IP instead of localhost.

You are making a tunnel that starts on local port ```$LPORT``` of your machine to destination port ```$IP:$PORT``` of the initial target ; then you'll need to use ```127.0.0.1:$LPORT``` to actually connect.

Reply:
Sure?, because it works as it is,

is not that localhost is where the remote IP where the PORT should be transmited?

So is from my localhost --> his localhost:PORT?

And I could say my localhost:$LPORT --> 127.0.1.10:$PORT? saying that I want to send my LPORT to his port but in that address?

Just asking

Reply:
But I understand that maybe (do not know) if the other machine is not correctly configured and can not resolve local host, the safest way would be LPORT:127.0.0.1:PORT

Is that what you mean?

Reply:
Yes sure.  It will only work if your target is the same as your SSH gateway but if you are using A as connection target and B as SSH gateway, it will not work.

Reply:
Sorry Robber Duck, think I got it

Reply:
Np, one of my [favorite article](https://blog.codinghorror.com/rubber-duck-problem-solving/), btw :-)

Reply:
All this should be fixed.